CorsixTH/Lua/app.lua

```diff
@@ -1476,33 +1476,33 @@ function App:getVersion(version)
   if ver > 156 then
     return "Trunk"
   elseif ver > 138 then
-    return "v0.65"
+    return "v0.65.0" -- Some users had v0.65.1 but no version number was specified
   elseif ver > 134 then
-    return "v0.64"
+    return "v0.64.0"
   elseif ver > 127 then
-    return "v0.63"
+    return "v0.63.0"
   elseif ver > 122 then
-    return "v0.62"
+    return "v0.62.0"
   elseif ver > 111 then
-    return "v0.61"
+    return "v0.61.0"
   elseif ver > 105 then
-    return "v0.60"
+    return "v0.60.0"
   elseif ver > 91 then
-    return "0.50"
+    return "v0.50.0"
   elseif ver > 78 then
-    return "0.40"
+    return "v0.40.0"
   elseif ver > 72 then
-    return "0.30"
+    return "v0.30.0"
   elseif ver > 66 then
-    return "0.21"
+    return "v0.21.0"
   elseif ver > 54 then
-    return "0.20"
+    return "v0.20.0"
   elseif ver > 53 then
-    return "0.11"
+    return "v0.11.0"
   elseif ver > 51 then
-    return "0.10"
+    return "v0.10.0"
   elseif ver > 45 then
-    return "0.01"
+    return "v0.01.0"
```

Comment:
Wouldn't it be simpler to have a table with numbers like `{ .... {53, "v0.11.0"}, {51, "v0.10.0"}, {45, "v0.01.0"} {-1, "Beta 8".}` and a loop finding the last item that matches?

Reply:
I can only guess the current system was made to failsafe savegame versions from matching an exact number.
So you'd need an error or fallback along with the for loop.

Reply:
Actually don't we have `rangeMapLookup` for this case I just described?

Reply:
I think the first person started with an `if`, and then everybody added another `elseif` at the end without ever considering alternatives.

Reply:
I don't think we really need it at all...